crates/host/src/wasmbus/mod.rs

```diff
@@ -1175,17 +1216,45 @@ impl ActorInstance {
             ..
         }: async_nats::Message,
     ) {
-        let res = self.handle_call(payload).await;
-        match (reply, res) {
-            (Some(reply), Ok(buf)) => {
-                if let Err(e) = self.nats.publish(reply, buf).await {
-                    error!("failed to publish response to `{subject}` request: {e:?}");
+        match rmp_serde::from_slice::<Invocation>(payload.as_ref()) {
+            Ok(invocation) => {
+                let invocation_id = invocation.id.clone();
+                attach_span_context(&invocation);
+                let res = self.handle_call(invocation).await;
+
+                let inv_resp = match res {
+                    Ok((msg, content_length)) => InvocationResponse {
+                        msg,
+                        invocation_id,
+                        content_length: content_length as u64,
```

Comment:
Don't cast, convert via `try_into`/`try_from` (this will be redundant if `handle_call` returns a `u64`)